src/main/java/com/launchacademy/reviews/controllers/VenueController.java

```diff
@@ -89,4 +85,20 @@ public VenueController(VenueService venueService,
       return new ResponseEntity<Object>(null, HttpStatus.UNPROCESSABLE_ENTITY);
     }
   }
+
+  @DeleteMapping("/{id}")
+    public ResponseEntity<Object> deleteVenue(@PathVariable Integer id){
+      Optional<Venue> venueToDelete = venueService.findById(id);
+
+      if(venueToDelete.isPresent()){
+        for(Review review : venueToDelete.get().getReviews()){
+          reviewService.delete(review);
+        }
+        venueService.delete(venueToDelete.get());
+        return new ResponseEntity<>(id, HttpStatus.OK);
```

Comment:
Nice work here! Nothin that you have to change, but you can drastically refactor this by adding in `orphanRemoval` to the model annotation.  Right now this is fine but may run into issues as the database grows and it needs to delete items one by one.

https://thorben-janssen.com/best-practices-many-one-one-many-associations-mappings/

Reply:
Added this change!